### DISCUSSION TITLE
`new Hono<{ Bindings: Env }>();` seems incompatible with Cloudflare wrangler generated type `Env`

### DISCUSSION THREAD

**Discussion by GregoryZeng** [honojs/hono#3080] [Q&A] [answered]
Hi, I am starting to use Hono with Cloudflare Workers. 
I find that `const app = new Hono<{ Bindings: Env }>();` would raise TypeScript error:
```
Type '{ Bindings: Env; }' does not satisfy the constraint 'Env'.
  Types of property 'Bindings' are incompatible.
    Type 'Env' is not assignable to type 'Bindings'.
      Index signature for type 'string' is missing in type 'Env'.ts(2344)
```
where `Env` is a type for Cloudflare's related bindings. It is in `./worker-configuration.d.ts` auto generated by command `wrangler types`:
```
interface Env {
	MY_VARIABLE: 'production_value';
	MY_BUCKET: R2Bucket;
	MY_DB: D1Database;
}
```
The TypeScript type inference works fine after I manually replace `interface Env` by `type Env = `. I am not expert in TypeScript typing and it is obviously not optimal.

Do you have any idea for this problem?

**Comment by yusukebe** [comment, marked as answer]
Hi @GregoryZeng 

> Do you have any idea for this problem?

Below are my ideas.

1. Don't use an `interface`. Use a `type` instead of it.
2. You can refer to the following code:

```ts
interface Env {
  MY_VARIABLE: 'production_value'
  MY_BUCKET: R2Bucket
  MY_DB: D1Database
}

type Bindings = {
  [key in keyof Env]: Env[key]
}

const app = new Hono<{ Bindings: Bindings }>()
```

**Reply by GregoryZeng** [reply]
Thanks! `[key in keyof Env]: Env[key]` is exactly what I need!
